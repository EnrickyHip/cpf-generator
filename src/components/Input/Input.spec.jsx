import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import { Input } from "."

describe("<Input />", () => {

  it("should call handleChange on each key pressed", () => {
    const text = 'value'
    const fn = jest.fn();

    render(<Input handleChange = {fn} />);

    const input = screen.getByPlaceholderText(/digite um CPF/i);
    userEvent.type(input, text);

    expect(fn).toBeCalledTimes(5);
  })

  /* ------------------------------------------------------- */

  it('should render with the value "123.456.789-00"', () => {
    const fn = jest.fn();
    
    render(<Input cpfInput = "123.456.789-00" handleChange = {fn} />);

    const input = screen.getByDisplayValue('123.456.789-00');

    expect(input).toBeInTheDocument;
  })

  it('should match snapshot', () => {
    const fn = jest.fn();
    const {container} = render(<Input cpfInput = "123.456.789-00" handleChange = {fn}/>);

    expect(container.firstChild).toMatchSnapshot();
  })

})