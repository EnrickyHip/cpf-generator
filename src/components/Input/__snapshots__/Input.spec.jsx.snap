// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<input
  autocomplete="off"
  class="full-input"
  id="cpf-input"
  maxlength="14"
  placeholder="Digite um CPF"
  value="123.456.789-00"
/>
`;
